aten/src/ATen/native/cuda/ForeachUnaryOp.cu

```diff
@@ -247,14 +247,15 @@ struct Abs {
 
 std::vector<Tensor> foreach_tensor_abs_cuda(TensorList tensors) {
     check_foreach_api_restrictions(tensors);
-    bool has_complex = false;
-    for (auto t : tensors) {
+    bool has_complex_or_integer = false;
+    for (const auto& t : tensors) {
         if (at::isComplexType(t.scalar_type())) {
-            has_complex = true;
+            has_complex_or_integer = true;
```

Comment:
this is misleading - how did `integer` get in the name? Also, std::any_of.

Reply:
ah, sorry, forgot to fix the name